tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -55,19 +54,22 @@ public void httpHandler(HttpServletRequest hsreq, HttpServletResponse hsres) {
 			
 			// 
 			if(request != null) {
+				WFSLogger.logInfo("WFS Server received GetCapabilities request.");
 				if(request.equals(WFSRequestType.GetCapabilities.toString())) {
 					getCapabilities(wr, hsreq);
 				}
 			}
 			
 			else{
-				wr.append("ncWFS on THREDDS - ");
-				wr.append(hsreq.getScheme() + "://" + hsreq.getServerName() + ":" + hsreq.getServerPort() + "/thredds/wfs");
+				WFSExceptionWriter owsExcept = new WFSExceptionWriter("WFS server error. REQUEST parameter is required.", "request", "MissingParameterValue", hsres);
+				owsExcept.write();
+				return;
 			}
 		}
 		
 		catch(IOException io) {

```

Comment:
Reading through log messages to diagnose a problem after it has happened is hard. It is easier if the same problem is only logged once. One widely-adopted convention that ensures a single source of truth in the event of a failure is to either rethrow an exception, or* to log it, but not both. Which do you think would be more appropriate here? Logging the exception or rethrowing the exception?

*"or" as in XOR.

Reply:
Yes, SLF4J shall be put on hold for the moment. Just will throw the single RuntimeException to compensate for this.